tests/Parallel_FilenameTest.cpp

```diff
@@ -0,0 +1,199 @@
+    MPI_Comm_size(MPI_COMM_WORLD, &mpiSize);
+
+    CPPUNIT_ASSERT(mpiSize == (MPI_SIZE_X * MPI_SIZE_Y));
+
+    dataCollector = new ParallelDataCollector(MPI_COMM_WORLD, MPI_INFO_NULL,
+            Dimensions(MPI_SIZE_X, MPI_SIZE_Y, 1), 1);
+}
+
+FilenameTest::~FilenameTest()
+{
+    if (dataCollector)
+    {
+        dataCollector->finalize();
+        delete dataCollector;
+        dataCollector = NULL;
+    }
+    if(!--initCt)
+        MPI_Finalize();
+}
+
+bool FilenameTest::fileExists(const std::string& filename)
+{
+    struct stat fileInfo;
+    return (stat(filename.c_str(), &fileInfo) == 0);
+}
+
+void FilenameTest::testBaseName()
+{
+    // MPI rank should be appended
+    runTest(HDF5_FILE, HDF5_BASE_FILE);
+}
+
+void FilenameTest::testFullName()
+{
+    // Filename should be used unchanged
+    runTest(HDF5_FULL_FILE, HDF5_FULL_FILE);
+}
+
+void FilenameTest::runTest(const char* filename, const char* fullFilename)
+{
+    CPPUNIT_ASSERT(!fileExists(fullFilename));
+
+    DataCollector::FileCreationAttr attr;
+    DataCollector::initFileCreationAttr(attr);
+    attr.fileAccType = DataCollector::FAT_CREATE;
+
+    // write first dataset to file (create file)
+    dataCollector->open(filename, attr);
+    int data1 = rand();
+
+    dataCollector->write(2, ctInt, 1, Selection(Dimensions(1, 1, 1)), "data1", &data1);
+    dataCollector->close();
+
+    // Now file must exist
+    CPPUNIT_ASSERT(fileExists(fullFilename));
+
+    // write second dataset to file (write to existing file of same name
+    attr.fileAccType = DataCollector::FAT_WRITE;
+    dataCollector->open(filename, attr);
+    int data2 = rand();
+
+    dataCollector->write(2, ctInt, 1, Selection(Dimensions(1, 1, 1)), "data2", &data2);
+    dataCollector->close();
+
+
+    // read data from file
+    attr.fileAccType = DataCollector::FAT_READ;
+    Dimensions data_size;
+
+    int data = -1;
+    dataCollector->open(filename, attr);
+
+    dataCollector->read(2, "data1", data_size, &data);
+
+    CPPUNIT_ASSERT(data_size.getScalarSize() == 1);
+    CPPUNIT_ASSERT(data == data1);
+
+    dataCollector->read(2, "data2", data_size, &data);
+
+    CPPUNIT_ASSERT(data_size.getScalarSize() == 1);
+    CPPUNIT_ASSERT(data == data2);
+
+    dataCollector->close();
+
+    // erase file
+    attr.fileAccType = DataCollector::FAT_CREATE;
+    dataCollector->open(filename, attr);
+
+    CPPUNIT_ASSERT_THROW(dataCollector->read(2, "data1", data_size, &data), DCException);
+    int data3 = rand();
+    dataCollector->write(2, ctInt, 1, Selection(Dimensions(1, 1, 1)), "data1", &data3);
+    dataCollector->close();
+
+
+    data = -1;
+    // Read from created file
+    attr.fileAccType = DataCollector::FAT_READ;
+    dataCollector->open(filename, attr);
+
+    dataCollector->read(2, "data1", data_size, &data);
+
+    CPPUNIT_ASSERT(data_size.getScalarSize() == 1);
+    CPPUNIT_ASSERT(data == data3);
+    dataCollector->close();
+
+    // Cleanup
+    unlink(fullFilename);
```

Comment:
To avoid these tests to influence each other. In these tests it should be made sure, that the correct files are created. Hence the "wrong" files should not exists. I could however introduce distinct base names for the files, if you like.